Create api for EAP create, update, get

Rup-Narayan-Rajbanshi opened the pull request:
Addresses EAP

## Changes

- Modified models
- Added serializers
- Added api for create , update and get for eap form.
- Added test cases, factories
- Added Models for EAP activation
- Modified api for (create, update, get) field-report to accept EAP activation.
- Modified existing test case of field-report to accept EAP Activation.
- Added Models for EAP Final Report
- Added Api for EAP  Final Report
- Added Test case for EAP Final Report

- [x] Updated/ran unit tests
- [ ] Updated CHANGELOG.md

## Release

If there is a version update, make sure to tag the repository with the latest version.

Rup-Narayan-Rajbanshi committed:
Crate api for eap create, update, get

- Modified models
- Added serializers
- Added views
- Added test cases, factories

szabozoltan69 commented:
@thenav56 Please do the merge yourself when needed. I have a feeling that it is not so easy to remove an already-done migration step, but I may be wrong about that.

Rup-Narayan-Rajbanshi commented:
> @thenav56 Please do the merge yourself when needed. I have a feeling that it is not so easy to remove an already-done migration step, but I may be wrong about that.

@szabozoltan69 , sorry it was a mistake, Now there is a new migration file.

Rup-Narayan-Rajbanshi committed:
Add api for eap options

- removed api for indicator, status
- created a model for prioritized_risk for early actions
- modified test cases to accept many prioritized risks

Rup-Narayan-Rajbanshi committed:
Modified create, update, get of field report api to accept eap activation:

Rup-Narayan-Rajbanshi committed:
Modified test case for field report to check EAP Activation Data

- Solve issue for empty document for eap activation create and update

Rup-Narayan-Rajbanshi committed:
Add new choice field EAP Activation in field report model

Rup-Narayan-Rajbanshi committed:
Add models for EAP Activation Report

Rup-Narayan-Rajbanshi committed:
Added create, update, get api for EAP final report

Rup-Narayan-Rajbanshi committed:
Add test case for EAP final report

Rup-Narayan-Rajbanshi committed:
Fix error for districs_details, indicator_display, document_details for
EAP

Rup-Narayan-Rajbanshi committed:
Fix issue for delay time for eap list

szabozoltan69 commented:
Dear @Rup-Narayan-Rajbanshi , @frozenhelium – could we please merge this PR to develop? Even if not ready fully?
There are some EAP changes already in develop, and without these last changes I can not merge develop to my Django 3.2 branch (due to the above mentioned enum issue, which is solved since then in this branch).

Rup-Narayan-Rajbanshi commented:
> Dear @Rup-Narayan-Rajbanshi , @frozenhelium – could we please merge this PR to develop? Even if not ready fully? There are some EAP changes already in develop, and without these last changes I can not merge develop to my Django 3.2 branch (due to the above mentioned enum issue, which is solved since then in this branch).

No it is not reviewd so I wont recommend you to merge into develop. Regarding the above mentioned  enum issue I will pull the develop branch , solve the above enum issue and then push it and  you can merge it into develop. Is it okay ?? @szabozoltan69

Rup-Narayan-Rajbanshi committed:
Optimized query number for eap list

Rup-Narayan-Rajbanshi commented:
> Dear @Rup-Narayan-Rajbanshi , @frozenhelium – could we please merge this PR to develop? Even if not ready fully? There are some EAP changes already in develop, and without these last changes I can not merge develop to my Django 3.2 branch (due to the above mentioned enum issue, which is solved since then in this branch).

It is not reviewd so I won't recommend you to merge into develop. Regarding the above mentioned enum issue I will pull the develop branch , solve the above enum issue and then push it and then you can merge it into develop. @szabozoltan69

szabozoltan69 commented:
It is not super urgent. Take your time, even maybe the review step is ok for me.

Rup-Narayan-Rajbanshi committed:
Removed html form form the browsable api

Rup-Narayan-Rajbanshi committed:
Added caption for file upload

szabozoltan69 commented:
Is this branch still needed?

Rup-Narayan-Rajbanshi commented:
@szabozoltan69 ,Let it be in Draft.

tnagorra commented:
@Rup-Narayan-Rajbanshi @thenav56 Can we close this issue?
